### PR TITLE
fix: allow webp files on newdot

### PR DESCRIPTION
**Pull request by allroundexperts**
### Details

This PR adds `webP` as an accepted image format on New Dot.

### Fixed Issues

$ https://github.com/Expensify/App/issues/15565   
PROPOSAL: https://github.com/Expensify/App/issues/15565#issuecomment-1452349728

### Tests

1. Save a webp format image.
2. Open the app, open any chat
3. Press Add attachment and send the file
4. Verify that the file uploads without any error.

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

1. Save a webp format image.
2. Open the app, open any chat
3. Press Add attachment and send the file
4. Verify that the file uploads without any error.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/30054992/227366283-1a19ca7b-2c8e-4a1d-a84d-ea0ccd505d57.mp4

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/30054992/227366351-bea4d3a7-bfc9-434d-946b-130b4c0768f1.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/30054992/227366461-bed7a9ef-501a-49eb-8c32-ad25ca1d88b8.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/30054992/227366585-7375c623-2e7c-4f68-a6bc-ef238dee0ca8.mp4

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/30054992/227366883-8c013087-6f61-4ab9-bba2-cdcebb1a8b69.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/30054992/227367021-0dec5c08-7383-4cac-bf99-6f37db227d1e.mov

</details>

### CODE REVIEW

File: package-lock.json
Comment (Santhosh-Sellavel) [inline package-lock.json:49]: Why would we need this?
Your proposal doesn't give us any hint of using a third-party library

Comment (Santhosh-Sellavel) [reply package-lock.json:49]: Also, it's better to include why would we need these new dependencies in the PR description.

**Review by Santhosh-Sellavel — COMMENTED** [review]
@PauloGasparSv Are we good with the use of these libraries?

File: package-lock.json
Comment (allroundexperts) [reply package-lock.json:49]: Instead of using custom mappings for file signatures in our code, its better to use something that has been well tested and is maintained. The `magic-bytes` library has the mappings in it. The `Buffer` library is used to convert file content  (which is a string) to a `Buffer` such that the data can be compared with the mappings.

Comment (allroundexperts) [reply package-lock.json:49]: @Santhosh-Sellavel Updated the description.

Comment (Santhosh-Sellavel) [reply package-lock.json:49]: Cool thanks for updating, let's wait for @PauloGasparSv's thoughts too!

**Comment by PauloGasparSv** [comment]
Hey @allroundexperts (and cc @Santhosh-Sellavel) sorry about this, but we had another [internal discussion](https://expensify.slack.com/archives/C03TQ48KC/p1678993506321549) on this (originally to discuss the libs you added) and we decided that:

1. We should fix the `webp` file upload this time so it actually works instead of  validating the mimeType before uploading.
2. So we should make a change in the API and the App to allow the `webp` extension

@allroundexperts would you please remove the change we have now from this P.R. and implement the changes for allowing the `webp` file extension in newDot (I think the change should be done [here](https://github.com/Expensify/App/blob/194325b677465b91030b6b7b9666076d553c5972/src/CONST.js#L20))?

I'll change the API logic to allow `webp` in the meantime.

**Comment by allroundexperts** [comment]
> Hey @allroundexperts (and cc @Santhosh-Sellavel) sorry about this, but we had another [internal discussion](https://expensify.slack.com/archives/C03TQ48KC/p1678993506321549) on this (originally to discuss the libs you added) and we decided that:
> 
> 1. We should fix the `webp` file upload this time so it actually works instead of  validating the mimeType before uploading.
> 2. So we should make a change in the API and the App to allow the `webp` extension
> 
> @allroundexperts would you please remove the change we have now from this P.R. and implement the changes for allowing the `webp` file extension in newDot (I think the change should be done [here](https://github.com/Expensify/App/blob/194325b677465b91030b6b7b9666076d553c5972/src/CONST.js#L20))?
> 
> I'll change the API logic to allow `webp` in the meantime.

@PauloGasparSv Sure. I'll do that.

**Comment by allroundexperts** [comment]
@PauloGasparSv However, I think we should still validate the file types. As an example, consider someone uploading a mp3 file renamed as a `JPG`. I think that will throw the same error.

**Comment by PauloGasparSv** [comment]
> @PauloGasparSv However, I think we should still validate the file types. As an example, consider someone uploading a mp3 file renamed as a JPG. I think that will throw the same error.

I agree we should validate it in the client too @allroundexperts! But we should go with the decision of the team I think.

If this happens again we can start another discussion and mention this issue as a reason to implement the mimeType check (someone will probably find this while investigating if something happens too).

And it's good thing that the API will throw an error for invalid files even if the extension was changed, it confirms that it's ok to do this changes for now as we are validating the files somewhere.

Thks for the help and patience

**Comment by allroundexperts** [comment]
> > @PauloGasparSv However, I think we should still validate the file types. As an example, consider someone uploading a mp3 file renamed as a JPG. I think that will throw the same error.
> 
> I agree we should validate it in the client too @allroundexperts! But I think we should go with the decision of the team this time.
> 
> If this happens again we can start another discussion and mention this issue as a reason to implement the mimeType check (someone will probably find this while investigating if something happens).
> 
> And it's good thing that the API will throw an error for invalid files even if the extension was changed, it confirms that it's ok to do this changes for now as we are validating the files somewhere.
> 
> Thks for the help and patience

Sounds good. I'll revert and add the requested changes.

**Comment by PauloGasparSv** [comment]
BTW, I just finished the API P.R. but there is 1 problem that I'm trying to figure out:

image

Clicking on the `webp` image to view it works fine! But the image preview in the reportAction isn't showing up.

**Comment by allroundexperts** [comment]
> BTW, I just finished the API P.R. but there is 1 problem that I'm trying to figure out:
> 
> image
> 
> Clicking on the `webp` image to view it works fine! But the image preview in the reportAction isn't showing up.

Something might be wrong here:
https://github.com/allroundexperts/Expensify/blob/039db1fe75b1e2088f7a1997a935f66e7c928e51/src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.js#L11

However, this is just my guess. I can only test once the API changes are on staging. I've updated the PR with new code though.

**Comment by PauloGasparSv** [comment]
Thks  @allroundexperts, the problem was in the API side but that file helped me find it :D

I'm putting this on HOLD so we only merge this once the API P.R. is merged too! I already tested this on Web and will test on other platforms and sent the evidence here shortly.

**Comment by PauloGasparSv** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by PauloGasparSv** [comment]
I want make sure there isn't a way around not showing `webp` previews to include here, so testing that while reviewing here.

image

**Comment by PauloGasparSv** [comment]
> That is happening because of another repo code. This [check](https://github.com/Expensify/App/blob/7bfbb1f6b2eb895fff7f3692d1ee563d1a2232f3/src/components/AttachmentView.js#L84-L88) should render the ImageView but  Str.isImage doesn't return true for webp.
>
> I'll update that in the expensify-common repo here!

@allroundexperts would you mind if I put this on HOLD again until the Expensify-Common P.R. https://github.com/Expensify/expensify-common/pull/512 is merged?
Then we can pull from main again and update the version of Expensify-Common in `package-json`  making this P.R. show 
 previews for `webp` files : )

**Comment by PauloGasparSv** [comment]
The Expensify-Common P.R. https://github.com/Expensify/expensify-common/pull/512 was merged!

@allroundexperts can you please update the Expensify-Common version in [package.json](https://github.com/Expensify/Web-Expensify/blob/c331da64d790a25d600e42c50fe36a0a633c260c/package.json#L26) to the latest commit (similar to [here](https://github.com/Expensify/App/pull/15409))?
```diff
- "expensify-common": "git+ssh://git@github.com/Expensify/expensify-common.git#ce6258b87396d3c471c9b012b9a518e2ee6283c7",
+ "expensify-common": "git+ssh://git@github.com/Expensify/expensify-common.git#9cf047b9741d3c02820d515dccb9e0a45b6595f5",
```

**Review by PauloGasparSv — APPROVED** [review]
LGTM!

All yours @Santhosh-Sellavel

Files I used to test are in a Zip in the bottom of [this comment](https://github.com/Expensify/App/pull/16031#issuecomment-1479428139) if anyone needs it for testing iOS and Android

**Comment by Santhosh-Sellavel** [comment]
Will get it today!

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by Santhosh-Sellavel** [comment]
@PauloGasparSv I see a few console warnings while adding attachments, this should be related to change but just for logging that we found adding this here, let me know your thoughts.

Check between 40th to 60th second in the below video for logs

https://user-images.githubusercontent.com/85645967/227644049-e66fde50-236c-4e89-8059-796bdf11f394.mp4

**Comment by Santhosh-Sellavel** [comment]
Tested it out completely, LGTM mostly. But I ran into a weird case on iOS check the below video. While importing the pics you added for testing on the simulator Actual JPG, GIF, and Webp as JPG are treated as Images and got imported to Photos and webp is treated as doc got copied to files.

GIF & Actual JPEG from photos get uploaded successfully. But the webp as JPG is not getting picked via Photos. 

Actual Webp & Webp as JPG can be uploaded successfully when picked via Files

https://user-images.githubusercontent.com/85645967/227648280-f49f3a99-b328-4f97-bc0d-fc2a7efe1b44.mp4


Let me know if need to do something about this @PauloGasparSv 
> But the webp as JPG is not getting picked via Photos.

**Review by Santhosh-Sellavel — COMMENTED** [review]
LGTM Mostly just few doubts!

**Comment by PauloGasparSv** [comment]
> @PauloGasparSv I see a few console warnings while adding attachments, this should be related to change but just for logging that we found adding this here, let me know your thoughts.

Thks @Santhosh-Sellavel, re-testing this to check that console error/warning!

> Tested it out completely, LGTM mostly. But I ran into a weird case on iOS check the below video. While importing the pics you added for testing on the simulator Actual JPG, GIF, and Webp as JPG are treated as Images and got imported to Photos and webp is treated as doc got copied to files.

I think that's ok :D

**Comment by PauloGasparSv** [comment]
Re-tested Web and couldn't see any warnings.

I'll test iOS again and merge this if no warnings show up : )

**Comment by PauloGasparSv** [comment]
All right, couldn't replicate that warning so I'm going to merge this @Santhosh-Sellavel!

**Pull request merged**
